src/commands/force/org/open.ts

```diff
@@ -0,0 +1,101 @@
+import { Messages, Org, MyDomainResolver, SfdxError, sfdc } from '@salesforce/core';
+import { Env, toNumber, Duration } from '@salesforce/kit';
+import { openUrl } from '../../../shared/utils';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-org', 'open');
+export class OrgOpenCommand extends SfdxCommand {
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessage('examples').split(EOL);
+  public static readonly requiresUsername = true;
+  public static readonly flagsConfig: FlagsConfig = {
+    path: flags.string({
+      char: 'p',
+      description: messages.getMessage('cliPath'),
+      default: '%2Flightning%2Fsetup',
+      env: 'FORCE_OPEN_URL',
+      parse: (input) => encodeURIComponent(decodeURIComponent(input)),
+    }),
+    urlonly: flags.boolean({
+      char: 'r',
+      description: messages.getMessage('urlonly'),
+    }),
+  };
+
+  public async run(): Promise<OrgOpenOutput> {
+    const frontDoorUrl = await this.buildFrontdoorUrl();
+    const url = `${frontDoorUrl}&retURL=${this.flags.path as string}`;
+    const orgId = this.org.getOrgId();
+    const username = this.org.getUsername();
+    const output = { orgId, url, username };
+
+    if (new Env().getString('SFDX_CONTAINER_MODE') ? true : false) {
+      // instruct the user that they need to paste the URL into the browser
+      this.ux.styledHeader('Action Required!');
+      this.ux.log(messages.getMessage('containerAction', [orgId, url]));
+      return output;
+    }
+
+    this.ux.log(messages.getMessage('humanSuccess', [orgId, username, url]));
+
+    if (this.flags.urlonly) {
+      return output;
+    }
+    // we actually need to open the org
+    await this.checkLightningDomain(url);
+    await openUrl(url);
+    return output;
+  }
+
+  private async buildFrontdoorUrl(): Promise<string> {
+    await this.org.refreshAuth(); // we need a live accessToken for the frontdoor url
+    const conn = this.org.getConnection();
+    const accessToken = conn.accessToken;
+    const instanceUrl = this.org.getField(Org.Fields.INSTANCE_URL) as string;
+    const instanceUrlClean = instanceUrl.endsWith('/') ? instanceUrl.substr(0, instanceUrl.length) : instanceUrl;
+
+    return `${instanceUrlClean}/secur/frontdoor.jsp?sid=${accessToken}`;
+  }
+
+  private async checkLightningDomain(url: string): Promise<void> {
+    const domain = `https://${/https?:\/\/([^.]*)/.exec(url)[1]}.lightning.force.com`;
+    const timeout = new Duration(toNumber(new Env().getString('SFDX_DOMAIN_RETRY', '240')), Duration.Unit.SECONDS);
+    if (sfdc.isInternalUrl(domain) || timeout.seconds === 0) {
+      return;
+    }
+
+    const resolver = await MyDomainResolver.create({
+      url: new URL(domain),
+      timeout,
+      frequency: new Duration(1, Duration.Unit.SECONDS),
+    });
+    this.ux.startSpinner(messages.getMessage('domainWaiting'));
```

Comment:
Do we need to stop it?

Reply:
It happens automatically when the command exits, right?  I see a lot of empty spinner-stopping code in the CLI and have been wondering why we do that.

Normally I only stop it if I want to say something specific in the message instead of the default "done".

Or because you think the thrown message before done makes people feel like it shouldn't say "done"?

Reply:
I ran into a weird situation in a fancy test where not stopping the spinner caused stdout not to be collected, causing a false test failure. Not sure why.

Reply:
weird.  I had a test for stdout in non-json mode, and just added one for stderr.  Both are fine without adding the extra stopspinner code.